agent/credentials/manager.go

```diff
@@ -123,12 +123,16 @@ func (manager *credentialsManager) SetTaskCredentials(taskCredentials TaskIAMRol
 }
 
 // GetTaskCredentials retrieves credentials for a given credentials id
-func (manager *credentialsManager) GetTaskCredentials(id string) (*TaskIAMRoleCredentials, bool) {
+func (manager *credentialsManager) GetTaskCredentials(id string) (TaskIAMRoleCredentials, bool) {
 	manager.taskCredentialsLock.RLock()
 	defer manager.taskCredentialsLock.RUnlock()
 
 	taskCredentials, ok := manager.idToTaskCredentials[id]
-	return taskCredentials, ok
+
+	if !ok {
+		return TaskIAMRoleCredentials{}, ok
```

Comment:
Without this change, if the `id` isn't found there will be a `*TaskIAMRoleCredentials` pointer instead of an empty `TaskIAMRoleCredentials` struct.